js/renovation/ui/toolbar/__tests__/toolbar.test.tsx

```diff
@@ -11,35 +12,95 @@ import LegacyToolbar from '../../../../ui/toolbar';
 describe('Toolbar', () => {
   describe('View', () => {
     it('default render', () => {
-      const buttonItem = new ToolbarItem();
-      buttonItem.options = new ToolbarButtonProps();
+      const toolbarProps = new ToolbarProps();
+      const initialProps = {
+        props: toolbarProps,
+        restAttributes: { 'rest-attributes': 'true' },
+      } as Partial<Toolbar>;
+      const tree = shallow(<ToolbarView {...initialProps as any} /> as any);
 
-      const buttonGroupItem = new ToolbarItem();
-      buttonGroupItem.options = new ToolbarButtonGroupProps();
+      expect(tree.find(DomComponentWrapper).props()).toMatchObject({
+        componentType: LegacyToolbar,
+        'rest-attributes': 'true',
+      });
+    });
 
-      const dropDownButtonItem = new ToolbarItem();
-      dropDownButtonItem.options = new ToolbarDropDownButtonProps();
+    it('pass items as string', () => {
+      const toolbarProps = new ToolbarProps();
+      toolbarProps.items = ['item1', 'item2'];
 
-      const checkBoxItem = new ToolbarItem();
-      checkBoxItem.options = new ToolbarCheckBoxProps();
+      const viewModel = new Toolbar(toolbarProps);
+      const tree = mount(ToolbarView(viewModel));
 
-      const textBoxItem = new ToolbarItem();
-      textBoxItem.options = new ToolbarTextBoxProps();
+      const resultProps = tree.find(DomComponentWrapper).props().componentProps;
+      expect(resultProps.items[0]).toEqual('item1');
+      expect(resultProps.items[1]).toEqual('item2');
+    });
 
-      const componentProps = new ToolbarProps();
-      componentProps.items = [buttonItem, buttonGroupItem,
-        dropDownButtonItem, checkBoxItem, textBoxItem];
+    it('Not crashed if items is undefined', () => {
+      const toolbarProps = new ToolbarProps();
 
-      const props = {
-        props: componentProps,
-        restAttributes: { 'rest-attributes': 'true' },
-      } as Partial<Toolbar>;
-      const tree = shallow(<ToolbarView {...props as any} /> as any);
+      const viewModel = new Toolbar(toolbarProps);
+      const tree = mount(ToolbarView(viewModel));
 
-      expect(tree.find(DomComponentWrapper).props()).toMatchObject({
-        componentProps,
-        componentType: LegacyToolbar,
-        'rest-attributes': 'true',
+      const resultProps = tree.find(DomComponentWrapper).props().componentProps;
+      expect(resultProps.items).toEqual(undefined);
+    });
+
+    each([false, true, undefined]).describe('rtlEnabled: %o', (isRtlEnabled) => {
+      it('correctly pass rtlEnabled', () => {
+        const buttonItem = new ToolbarItem();
+        const buttonGroupItem = new ToolbarItem();
+        const dropDownButtonItem = new ToolbarItem();
+        const checkBoxItem = new ToolbarItem();
+        const textBoxItem = new ToolbarItem();
+        const toolbarProps = new ToolbarProps();
+        toolbarProps.items = [buttonItem, buttonGroupItem,
+          dropDownButtonItem, checkBoxItem, textBoxItem];
```

Comment:
It's easier to read a flat array:
[new ToolbarItem(), new ToolbarItem(), new ToolbarItem(), new ToolbarItem(), new ToolbarItem()]

Reply:
Yep, great idea. Fixed

Reply:
Sorry, and regarding this question https://github.com/DevExpress/DevExtreme/pull/19003#discussion_r701236622
I cant reply directly to this question, so I reply here:
- fixed. Rename rtl to resolvedRtlEnabled, because it resolved from multiple sources